src/Bicep.Core/Semantics/LocalSymbol.cs

```diff
@@ -0,0 +1,21 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using Bicep.Core.Syntax;
+using Bicep.Core.TypeSystem;
+
+namespace Bicep.Core.Semantics
+{
+    public class LocalSymbol : DeclaredSymbol
+    {
+        public LocalSymbol(ISymbolContext context, string name, LocalVariableSyntax declaringSyntax)
```

Comment:
Why differentiate LocalSymbol from e.g. VariableSymbol?

Reply:
My thinking here was that they have different semantics. VariableSymbols tend to produce variable declarations in the JSON (unless inlined). Locals in loops will be replaced with some `copyIndex()` function call during codegen.

In the next PR I will be adding the code gen part. If I can get rid of it I will, but I will need to differentiate between loop vars and top-level vars somehow. Btw Roslyn does have `SymbolKind.Local` but I was hoping to get rid of all the Kind enums entirely at some point.